store/tikv/gc_worker.go

```diff
@@ -36,17 +36,48 @@ import (
 
 // GCWorker periodically triggers GC process on tikv server.
 type GCWorker struct {
-	uuid        string
-	desc        string
-	store       *tikvStore
-	gcIsRunning int32
+	uuid  string
+	desc  string
+	store *tikvStore
+
+	/*
+	 * gc worker uses "session"
+	 * safepoint checker uses "spSession"
+	 * so there is no race
+	 */
+	session   tidb.Session
+	spSession tidb.Session
+
+	gcIsRunning bool
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
 }
 
+// StartSafePointChecker triggers SafePoint Checker on each tidb
+func (w *GCWorker) StartSafePointChecker() {
```

Comment:
I doubt that operating safe point in two different workers can't guarantee data consistency.